src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -276,15 +279,17 @@ private void createInferenceListForMapTypeInput(Object sourceValue, List<String>
     }
 
     @VisibleForTesting
-    Map<String, Object> buildMapWithTargetKeyAndOriginalValue(IngestDocument ingestDocument) {
+    Map<String, Object> buildMapWithTargetKeys(IngestDocument ingestDocument) {
         Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
         Map<String, Object> mapWithProcessorKeys = new LinkedHashMap<>();
         for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
-            String originalKey = fieldMapEntry.getKey();
-            Object targetKey = fieldMapEntry.getValue();
+            Pair<String, Object> processedNestedKey = processNestedKey(fieldMapEntry);
```

Comment:
Is the fieldMapEntry what user define during processor creation?
If that is the case, can we parse and expand the key during creation time?

Reply:
yes, it should be possible to construct that map when we create a processor class. let me do the change

Reply:
Cool. That way, we might be able to reduce some latency during ingestion.